cli/inspector.rs

```diff
@@ -0,0 +1,200 @@
+use crate::version::DENO;
+use futures::sync::oneshot::{spawn, SpawnHandle};
+use futures::{Future, Sink, Stream};
+use std::net::SocketAddrV4;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::{Arc, Mutex};
+use warp::ws::{Message, WebSocket};
+use warp::Filter;
+
+pub struct Inspector {
+  enable: bool,
+  address: SocketAddrV4,
+  // sharable handle to channels passed to isolate
+  pub handle: deno::InspectorHandle,
+  // sending/receving messages from isolate
+  inbound_tx: Arc<Mutex<Sender<String>>>,
+  outbound_rx: Arc<Mutex<Receiver<String>>>,
+  // signals readiness of inspector
+  // TODO(mtharrison): Is a condvar more appropriate?
+  ready_tx: Arc<Mutex<Sender<()>>>,
+  ready_rx: Arc<Mutex<Receiver<()>>>,
+  server_spawn_handle: Option<SpawnHandle<(), ()>>,
+  // TODO(mtharrison): Maybe use an atomic bool instead?
+  started: Arc<Mutex<bool>>,
+  connected: Arc<Mutex<bool>>,
+}
+
+impl Inspector {
+  pub fn new(enable: bool, address: Option<String>) -> Self {
+    let (inbound_tx, inbound_rx) = channel::<String>();
+    let (outbound_tx, outbound_rx) = channel::<String>();
+    let (ready_tx, ready_rx) = channel::<()>();
+
+    let address = match address {
+      Some(address) => address.parse::<SocketAddrV4>().unwrap(),
```

Comment:
If malformed address is passed then the process will panic and die. How about chaning signature to `pub fn new(enable: bool, address: Option<String>) -> Result<Self, Errbox>` and handling those unwraps like so: `address.parse::<SocketAddrV4>()?`?